pkg/agent/datapath/utils.go

```diff
@@ -143,27 +143,45 @@ func datapathRule2RpcRule(entry *EveroutePolicyRuleEntry) *v1alpha1.RuleEntry {
 }
 
 func (rule EveroutePolicyRule) MatchConntrackFlow(flow *netlink.ConntrackFlow) bool {
+	forwardMatch, reverseMatch := true, true
+
 	if rule.IPProtocol != 0 && rule.IPProtocol != flow.Forward.Protocol {
-		return false
+		forwardMatch = false
 	}
-
 	if rule.SrcIPAddr != "" && !matchIP(rule.SrcIPAddr, flow.Forward.SrcIP) {
-		return false
+		forwardMatch = false
 	}
-
 	if rule.DstIPAddr != "" && !matchIP(rule.DstIPAddr, flow.Forward.DstIP) {
-		return false
+		forwardMatch = false
 	}
-
 	if rule.SrcPort != 0 && !matchPort(rule.SrcPortMask, rule.SrcPort, flow.Forward.SrcPort) {
-		return false
+		forwardMatch = false
 	}
-
 	if rule.DstPort != 0 && !matchPort(rule.DstPortMask, rule.DstPort, flow.Forward.DstPort) {
-		return false
+		forwardMatch = false
+	}
+
+	if forwardMatch {
+		return true
+	}
+
+	if rule.IPProtocol != 0 && rule.IPProtocol != flow.Reverse.Protocol {
+		reverseMatch = false
+	}
+	if rule.SrcIPAddr != "" && !matchIP(rule.SrcIPAddr, flow.Reverse.SrcIP) {
+		reverseMatch = false
+	}
+	if rule.DstIPAddr != "" && !matchIP(rule.DstIPAddr, flow.Reverse.DstIP) {
+		reverseMatch = false
+	}
+	if rule.SrcPort != 0 && !matchPort(rule.SrcPortMask, rule.SrcPort, flow.Reverse.SrcPort) {
+		reverseMatch = false
+	}
+	if rule.DstPort != 0 && !matchPort(rule.DstPortMask, rule.DstPort, flow.Reverse.DstPort) {
+		reverseMatch = false
```

Comment:
nit: pulling out the duplicate code as a separate method.
e.g.
```
func (rule EveroutePolicyRule) MatchConntrackFlow(flow *netlink.ConntrackFlow) bool {
	return rule.matchIPTuple(
		flow.Forward.Protocol,
		flow.Forward.SrcIP,
		flow.Forward.DstIP,
		flow.Forward.SrcPort,
		flow.Forward.DstPort,
	) ||
		rule.matchIPTuple(
			flow.Reverse.Protocol,
			flow.Reverse.SrcIP,
			flow.Reverse.DstIP,
			flow.Reverse.SrcPort,
			flow.Reverse.DstPort,
		)
}

func (rule EveroutePolicyRule) matchIPTuple(protocol uint8, srcIP, dstIP net.IP, srcPort, dstPort uint16) bool
```

Reply:
done